examples/client/wayland_client.c

```diff
@@ -47,30 +79,30 @@ static void new_global(
     char const* interface,
     uint32_t version)
 {
-    (void)version;
-    struct globals* globals = data;
+    (void)data;
 
     if (strcmp(interface, "wl_compositor") == 0)
     {
-        globals->compositor = wl_registry_bind(registry, id, &wl_compositor_interface, 3);
+        globals.compositor = wl_registry_bind(registry, id, &wl_compositor_interface, min(version, 3));
```

Comment:
Why are we accepting any version below here? Same question for other globals. Is it possible that a version isn't backwards compatible?

Reply:
We can't bind to a version higher than the version supported by the server. In theory, we need to avoid any requests that don't exist in the agreed version, but we are pretty conservative in what we use in the demo,